gazelle/gazelle.go

```diff
@@ -0,0 +1,258 @@
+			Imp: fmt.Sprintf("//%s:%s", f.Pkg, src),
+		}
+
+		imports = append(imports, spec)
+	}
+
+	return imports
+}
+
+// Embeds returns a list of labels of rules that the given rule embeds. If
+// a rule is embedded by another importable rule of the same language, only
+// the embedding rule will be indexed. The embedding rule will inherit
+// the imports of the embedded rule.
+// Since SkyLark doesn't support embedding this should always return nil.
+func (*bzlLibraryLang) Embeds(r *rule.Rule, from label.Label) []label.Label { return nil }
+
+// Resolve translates imported libraries for a given rule into Bazel
+// dependencies. Information about imported libraries is returned for each
+// rule generated by language.GenerateRules in
+// language.GenerateResult.Imports. Resolve generates a "deps" attribute (or
+// the appropriate language-specific equivalent) for each import according to
+// language-specific rules and heuristics.
+func (*bzlLibraryLang) Resolve(c *config.Config, ix *resolve.RuleIndex, rc *repo.RemoteCache, r *rule.Rule, importsRaw interface{}, from label.Label) {
+	imports := importsRaw.([]string)
+
+	r.DelAttr("deps")
+
+	if len(imports) == 0 {
+		return
+	}
+
+	deps := make([]string, 0, len(imports))
+	for _, imp := range imports {
+		if strings.HasPrefix(imp, "@") {
+			// This is a dependency that is external to the current repo.
+			deps = append(deps, strings.TrimSuffix(imp, fileType))
+		} else {
+			res := resolve.ImportSpec{
+				Lang: languageName,
+				Imp:  imp,
+			}
+			matches := ix.FindRulesByImport(res, languageName)
+
+			if len(matches) == 0 {
+				log.Printf("%s: %q was not found in dependency index. Skipping. This may result in an incomplete deps section and require manual BUILD file intervention.\n", from.String(), imp)
+			}
+
+			for _, m := range matches {
+				deps = append(deps, m.Label.String())
+			}
+		}
+	}
+
+	sort.Strings(deps)
+	if len(deps) > 0 {
+		r.SetAttr("deps", deps)
+	}
+}
+
+var kinds = map[string]rule.KindInfo{
+	"bzl_library": {
+		NonEmptyAttrs:  map[string]bool{"srcs": true, "deps": true},
+		MergeableAttrs: map[string]bool{"srcs": true},
+	},
+}
+
+// GenerateRules extracts build metadata from source files in a directory.
+// GenerateRules is called in each directory where an update is requested
+// in depth-first post-order.
+//
+// args contains the arguments for GenerateRules. This is passed as a
+// struct to avoid breaking implementations in the future when new
+// fields are added.
+//
+// A GenerateResult struct is returned. Optional fields may be added to this
+// type in the future.
+//
+// Any non-fatal errors this function encounters should be logged using
+// log.Print.
+func (*bzlLibraryLang) GenerateRules(args language.GenerateArgs) language.GenerateResult {
+	var rules []*rule.Rule
+	var imports []interface{}
+	for _, f := range append(args.RegularFiles) {
+		if ignoreSuffix.Matches(f) {
+			continue
+		}
+		if strings.HasSuffix(f, fileType) {
```

Comment:
Might be more readable to replace `ignoreSuffix` and `fileType` with a predicate:

```
func isBzlSourceFile(name string) bool { ... }
```

Alternatively, consolidate these two conditions and unindent the rest of the loop body:

```
if ignoreSuffix.Matches(f) || !strings.HasSuffix(f, fileType) {
  continue
}
```